File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.py
def solution(arr, n):
    l = len(arr)
    
    if l % 2 == 0:
        for i in range(1, l, 2):
            arr[i] = arr[i] + n
    else:
        for i in range(0, l, 2):
            arr[i] = arr[i] + n
    
    return arr

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    var answer = [...arr];

    
    if(answer.length % 2 == 0){
        answer = answer.map((x, i) => {
            return (i % 2 == 1) ? (x + n) : x;
        })
    }else{
        answer = answer.map((x, i) => {
            return (i % 2 == 0) ? (x + n) : x
        })
    }
    
    return answer;
}